src/Build/Collections/PropertyDictionary.cs

```diff
@@ -82,6 +84,15 @@ internal PropertyDictionary(IEnumerable<T> elements)
         }
 
         /// <summary>
+        /// Creates empty dictionary, specifying a comparer
+        /// </summary>
+        internal PropertyDictionary(MSBuildNameIgnoreCaseComparer comparer)
```

Comment:
Instead of sharing the same mutable instance, each request gets a new object. This avoids lock contention inside the comparer.